src/main/cpp/src/cast_string.cu

```diff
@@ -248,6 +680,70 @@ struct string_to_integer_impl {
   }
 };
 
+struct string_to_decimal_impl {
+  /**
+   * @brief create deimal column of type T from strings column
+   *
+   * @tparam T decimal type requested
+   * @param dtype data type of result column, includes scale
+   * @param precision precision of incoming string data
+   * @param string_col strings to convert to decimal
+   * @param ansi_mode strict ansi mode checking of incoming data, can throw
+   * @param stream stream on which to operate
+   * @param mr memory resource to use for allocations
+   * @return std::unique_ptr<column> decimal column created from strings
+   */
+  template <typename T, typename std::enable_if_t<is_fixed_point<T>()>* = nullptr>
+  std::unique_ptr<column> operator()(data_type dtype,
+                                     int32_t precision,
+                                     strings_column_view const& string_col,
+                                     bool ansi_mode,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr)
+  {
+    using Type = device_storage_type_t<T>;
+
+    rmm::device_uvector<Type> data(string_col.size(), stream, mr);
+    auto const num_words = bitmask_allocation_size_bytes(string_col.size()) / sizeof(bitmask_type);
+    rmm::device_uvector<bitmask_type> null_mask(num_words, stream, mr);
+
+    dim3 const blocks(util::div_rounding_up_unsafe(string_col.size(), detail::NUM_THREADS));
+    dim3 const threads{detail::NUM_THREADS};
+
+    detail::string_to_decimal_kernel<<<blocks, threads, 0, stream.value()>>>(
```

Comment:
Can we maximize code share between `string_to_integer_kernel` and `string_to_decimal_kernel`? Like extracting some helper functions that can be called from both places.

Reply:
That is 100% the goal here and I did pull out some common functions like `will_overflow`, `is_whitespace`, and `process_value`. I believe there is more sharing going on here and we can pull more out. My plan was to continue to factor out common code as more casts are added. There is a list we are going down for these.